src/Mvc/Collection.php

```diff
@@ -0,0 +1,1555 @@
+            }
+        }
+
+        $this->collectionsManager = $collectionsManager;
+
+        $collectionsManager->initialize($this);
+
+        if (method_exists($this, 'onConstruct')) {
+            $this->onConstruct($data);
+        }
+
+        if (is_array($data)) {
+            $this->assign($data);
+        }
+    }
+
+    /**
+     * @param BehaviorInterface $behavior
+     */
+    protected function addBehavior(BehaviorInterface $behavior): void
+    {
+        $this->collectionsManager->addBehavior($this, $behavior);
+    }
+
+    /**
+     * @param MessageInterface $message
+     * @return mixed|void
+     */
+    public function appendMessage(MessageInterface $message): CollectionInterface
+    {
+        $this->errorMessages[] = $message;
+
+        return $this;
+    }
+
+    /**
+     * Perform an aggregation using the Mongo aggregation framework
+     *
+     * @param array|null $parameters
+     * @param array|null $options
+     * @return array
+     * @throws Exception
+     */
+    public static function aggregate(?array $parameters = [], ?array $options = []): array
+    {
+        $className = get_called_class();
+        /** @var CollectionInterface $collection */
+        $collection = new $className();
+
+        $source = $collection->getSource();
+        if (empty($source)) {
+            throw new Exception("Method getSource() returns empty string");
+        }
+
+        $connection = $collection->getConnection();
+        $cursorOrArrayIterator = $connection->selectCollection($source)->aggregate($parameters, $options);
+
+        if ($cursorOrArrayIterator instanceof Cursor) {
+            return $cursorOrArrayIterator->toArray();
+        }
+
+        return (array)$cursorOrArrayIterator;
+    }
+
+    /**
+     * @param CollectionInterface|EntityInterface $base
+     * @param array $data
+     * @param int $dirtyState
+     * @return CollectionInterface
+     */
+    public static function cloneResult(CollectionInterface $base, array $data, int $dirtyState = 0): CollectionInterface
+    {
+        $collection = clone $base;
+
+        foreach ($data as $key => $value) {
+            $collection->writeAttribute($key, $value);
+        }
+
+        if (method_exists($collection, 'afterFetch')) {
+            $collection->afterFetch();
+        }
+
+        return $collection;
+    }
+
+    /**
+     * Creates a collection based on the values in the attributes
+     *
+     * @return bool
+     * @throws Exception
+     */
+    public function create(): bool
+    {
+        $collection = $this->prepareCU();
+
+        $exists = false;
+        $this->operationMade = self::OP_CREATE;
+        $this->errorMessages = [];
+
+        if ($this->preSave(self::$disableEvents, $exists) === false) {
+            return false;
+        }
+
+        $success = false;
+        $status = $collection->insertOne($this, [
+            'w' => true
```

Comment:
```suggestion
            'w' => true,
```

Reply:
Why ? There is only one parameter

Reply:
https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

Reply:
Thanks for the information. It's interesting :)
It's obviously interesting for the git history